### PR TITLE
Report error when a RealmObject class has no persisted properties

### LINKED ISSUE #689: Error when a RealmObject class has no persisted properties

### PR DESCRIPTION
**Pull request by kristiandupont**
This fixes #689.

### CODE REVIEW

**Comment by fealebenpae** [comment]
👍

File: Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs [hunk 1]
```diff
@@ -18,13 +18,16 @@
...
+using System.Configuration;
 using Realms;
 
 namespace AssemblyToProcess
 {
     public class RealmListWithSetter : RealmObject
     {
         public IList<Person> People { get; set; } 
+
+        public int PropertyToEnsureOtherwiseHealthyClass { get; set; }
```
Comment (AndyDentFree) [inline Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs]: This should not be necessary - `IList` alone is legal

File: Tests/WeaverTests/AssemblyToProcess/RemappedClass.cs [hunk 2]
```diff
@@ -28,5 +28,6 @@ namespace AssemblyToProcess
     [MapTo("RemappedTable")]
```
Comment (AndyDentFree) [inline Tests/WeaverTests/AssemblyToProcess/RemappedClass.cs]: are we dropping `MapTo`? If that is the case then we could  drop this entire test

File: Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs (hunk 1 above)
Comment (kristiandupont) [inline Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs]: Not when the `IList` has a `set` method.

File: Tests/WeaverTests/AssemblyToProcess/RemappedClass.cs (hunk 2 above)
Comment (kristiandupont) [inline Tests/WeaverTests/AssemblyToProcess/RemappedClass.cs]: We never really concluded anything, so I decided to leave it for now.

File: Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs (hunk 1 above)
Comment (AndyDentFree) [inline Tests/WeaverTests/AssemblyToProcess/FaultyClasses.cs]: doh!

**Comment by kristiandupont** [comment]
Retest this please

**Pull request merged**